content/docs/engine-dev/tools/tiaf/maintaining-tiaf/debugging-tiaf.md

```diff
@@ -77,47 +87,179 @@ In the log above, you can extract the following information about the AR run and
+    --sequence=tianowrite --fpolicy=continue --suites=main,smoke --labelexcludes=REQUIRES_gpu --ipolicy=continue --changelist=D:\workspace\o3de\build\windows\bin\TestImpactFramework\profile\Temp\Native\changelist.06011392e5a7456e817cda03f8f8c606.json --report=D:\workspace\o3de\build\windows\bin\TestImpactFramework\profile\Temp\Native\Reports\report.06011392e5a7456e817cda03f8f8c606.json --targetout=stdout
     ```
 
 To reproduce this particular AR run, perform the following steps:
 
-1. Check out the commit `59e24a62859ce8a932e338f36432e3228621c1ec`.
-2. Use the [storage query tool](./storage-query-tool.md) to retrieve the relevant historic data for this run.
-3. Build the commit.
-4. Place the historic data in the local historic data folder (see the historic workspace entry in any of the `tiaf.json` files in the TestImpactFramework folder in your build directory).
-5. Run the TIAF AR script with the the following arguments (omitting the S3 bucket so that local persistent storage is used):
+1. Recreate the historic data locally.
+2. Build and run the commit failing in AR.
+3. Run the TIAF runtime with a debugger attached.
+
+### Step 1: Recreate the historic data locally
+
+To recreate the historic data locally, you can either use the [storage query tool](./storage-query-tool.md) to retrieve the relevant historic data for this run from the server running AR and place the contents in the `<build_path>/bin/TestImpactFramework/debug/Persistent/Native/historic/main-smoke` directory, or recreate the historic data from scratch. As the latter approach is the most reliable (as it does not require access to the persistent storage used by AR), this is the approach that will be described below.
+
+Checkout and build the commit used by the last run (in this instance `4b65521188a2a0beb399eca7b692216d2fe3d209`). For native runs, it is recommended you build the `debug` configuration to make debugging easier but for Python runs you must build the `profile` configuration as, at the time of writing, Python tests will not execute properly with the `debug` build configuration.
+
+Run the TIAF using the AR scripts to generate the historic data used in the failing AR run locally. To do so, run the `tiaf_driver.py` script and supply the arguments used in the AR run. For this demonstration, Visual Studio Code will be used. 
+
+The script configuration file used for this step is as follows:
+
+```json
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```

Comment:
Just to verify that the json parser that the tiaf_driver uses handles these comments right?

Reply:
Sorry, I realise this is not exactly clear (it's the VS Code `launch.json`). I've clarified this and linked to the MS page describing this config file 👍